x-pack/plugins/case/server/client/cases/update.ts

```diff
@@ -126,6 +130,65 @@ export const update = ({
       }),
     });
 
+    // If a status update occurred and the case is synced then we need to update all alerts' status
+    // attached to the case to the new status.
+    const casesWithStatusChangedAndSynced = updateFilterCases.filter((caseToUpdate) => {
```

Comment:
Please add some tests around this function, a follow up PR would be fine 👍

Reply:
I will add tests for the whole feature (attach alert to a case) on a follow-up PR!